File: 1582-special-positions-in-a-binary-matrix/1582-special-positions-in-a-binary-matrix.py
class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        row=len(mat)
        col=len(mat[0])
        r=[0]*row
        c=[0]*col
        for i in range(row):
            for j in range(col):
                if mat[i][j]==1:
                    r[i]+=1
                    c[j]+=1
        count=0
        for i in range(row):
            for j in range(col):
                if mat[i][j]==1 and r[i]==1 and c[j]==1:
                    count +=1
        return count

File: 1582-special-positions-in-a-binary-matrix/1582-special-positions-in-a-binary-matrix.java
class Solution {
    public int numSpecial(int[][] mat) {
        int row=mat.length;
        int col=mat[0].length;
        int[] r=new int[row];
        int[] c=new int[col];
        for(int i=0;i<row;i++){
            for(int j=0;j<col;j++){
                if(mat[i][j]==1){
                    r[i]++;
                    c[j]++;
                }
            }
        }
        int count=0;
        for(int i=0;i<row;i++){
            for(int j=0;j<col;j++){
                if(mat[i][j]==1 && r[i]==1 && c[j]==1){
                    count++;
                }
            }
        }
        return count;
        
    }
}